internal/services/mssqlmanagedinstance/mssql_managed_instance_transparent_data_encryption_resource_test.go

```diff
@@ -440,13 +451,202 @@ resource "azurerm_mssql_managed_instance" "test" {
     database    = "test"
   }
 }
+`, db.template(data, data.Locations.Primary), data.RandomInteger)
+}
 
+func (r MsSqlManagedInstanceTransparentDataEncryptionResource) managedHSM(data acceptance.TestData) string {
+	db := MsSqlManagedInstanceResource{}
+	return fmt.Sprintf(`
+%s
 
+provider "azurerm" {
+  features {
+    key_vault {
+      purge_soft_delete_on_destroy       = false
+      purge_soft_deleted_keys_on_destroy = false
+    }
 
+    resource_group {
+      /* Due to the creation of unmanaged Microsoft.Network/networkIntentPolicies in this service, 
+      prevent_deletion_if_contains_resources has been added here to allow the test resources to be 
+       deleted until this can be properly investigated
+      */
```

Comment:
So that we can track and potentially add this to our planning, I think it would make sense to open an issue for this and have the issue link here as a comment. WDYT?